lua/treesitter-context.lua

```diff
@@ -529,10 +529,25 @@ local function highlight_contexts(bufnr, ctx_bufnr, contexts)
   end
 end
 
+--- @param win integer
 --- @param lnum integer
+--- @param relnum integer
 --- @param width integer
 --- @return string
-local function build_lno_str(lnum, width)
+local function build_lno_str(win, lnum, relnum, width)
+  if vim.wo[win].statuscolumn and vim.wo[win].statuscolumn ~= ""  then
+    vim.v.lnum = lnum
+    vim.v.relnum = relnum -- TODO: this variable is currently readonly
+    vim.g.statusline_winid = win
+    local ok, data = pcall(api.nvim_eval_statusline,vim.wo[win].statuscolumn, {
+      winid = win,
+      use_statuscol = true,
+      maxwidth = 0,
```

Comment:
This is also calculated accordingly (`win_col_off()`) and should be removed.

> the statuscolumn is initially missing some content, as the numbers are flush with the the rest of the content, but on moving my cursor the first time they get populated.

I'm not sure if that will fix this issue. Is `'statuscolumn'` set before this plugin loads in your config?

Reply:
@luukvbaal I believe you are right and it is a timing issue. AFAICT my statuscolumn should be coming in way before the context window, as that is loaded as a `plugin/statuscolumn`, I even tried delaying it to `CursorHold` but still seem that for whatever reason the context window starts first. Anyway this isn't relevant to your PR I think which is working nicely 👍🏾

Reply:
I guess it' probably because it's an nvim-treesitter module and so it is required quite early in the startup process and so actually runs before `plugin/statuscolumn` 🤷🏾‍♂️ (just thinking aloud not relate to your work @luukvbaal )